src/contract.rs

```diff
@@ -0,0 +1,151 @@
+use std::{
+    cell::RefCell,
+    collections::{BTreeMap, BTreeSet},
+};
+
+use nickel_lang_core::term::RichTerm;
+
+use crate::{references::AcyclicReferences, utils::static_access};
+
+/// The owned part of a [`ContractContext`].
+pub struct ContractContextData<'a, 'refs> {
+    refs: &'a AcyclicReferences<'refs>,
```

Comment:
The fields' meaning aren't totally obvious for me right now, so they might not be for someone that has never worked on json-schema-to-nickel. A short description could help!